src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -141,7 +145,7 @@ def attach_data(
         # set local properties on the model
         self._copy_trainer_model_properties(model)
 
-    def _copy_trainer_model_properties(self, model: "pl.LightningModule") -> None:
+    def _copy_trainer_model_properties(self, model: TypeAlias["pl.LightningModule"]) -> None:
```

Comment:
I don't think `TypeAlias` is appropriate - what are you trying to solve here?

Reply:
not able to find the original error in CI/CD checks; I removed locally and it is fine. Updated in https://github.com/Lightning-AI/lightning/pull/13806/commits/82ee684a0b76486eee08b0b9f7462762edf3ae51